### PR TITLE
Bazel builder for desktop targets

### PR DESCRIPTION
**Pull request by tanishqaggarwal**
### Summary of changes
- Added Bazel build support to the repository for native builds
- Streamlined testing by adding a `PAN_TEST()` macro that takes care of the `main()` boilerplate at the end of unit tests.
- GitHub builds based on Bazel
- Renamed `.enum` files to `.enum.h` (required for Bazel to work)
- Renamed `DownlinkProducer::flow_data` to `PAN::flow_data`

Up and coming work: get this to work for Teensy-based targets.

### Testing
Check out `.github/workflows/bazel.yml` for the set of targets you can try out right now.

### Constants
No important changes.

### Telemetry
No changes.

### Documentation Evidence
I'll create a README for how to use the Bazel invocations once I've finished the full set of work (including Arduino compilation.) For now, depend on the Github Workflows file for invocations.

### CODE REVIEW

**Comment by tanishqaggarwal** [comment]
@athena255 Do you still need propulsion_fault_t.enum? This PR deleted it.

**Comment by tanishqaggarwal** [comment]
I think it's important to double-check that the list of tests in tests/BUILD isn't missing any tests. I'll do this one more time, could you guys also scan through the list?

**Comment by tanishqaggarwal** [comment]
Closing to simplify the structure. There's way too much code repetition that can be simplified by using Bazel `select`s.

File: WORKSPACE [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+lin_build = """
+cc_library(
+    name = "lin",
+    srcs = glob(["src/**/*.cpp", "include/**/*.inl"]),
+    hdrs = glob(["include/**/*.hpp"]),
+    includes = ["include"],
+    linkstatic = True,
+    visibility = ["//visibility:public"]
+)
+"""
```
Comment (kkrol27) [inline WORKSPACE:13]: You shouldn't need to define this here as well. It should automatically find it in the lin repo itself.  I got this to work with my new PR in lin: https://github.com/pathfinder-for-autonomous-navigation/lin/pull/12

File: lib/common/BUILD [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    copts = ["-Ilib/common/libsbp/include/libsbp/ -Ilib/common/libsbp/include/"],
+    linkstatic = True,
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "psim",
+    srcs = glob(["psim/src/**/*.cpp", "psim/src/**/*.inl", "psim/src/**/*.hpp", "psim/include/**/*.inl"]),
+    hdrs = glob(["psim/include/**/*.hpp", "psim/include/**/*.h"]),
+    copts = ["-Ilib/common/psim/include -Ilib/common/psim/src"],
```
Comment (kkrol27) [inline lib/common/BUILD:37]: Can we not add these as includes? Also, `src` shouldn't be an include here.

File: lib/fsw/BUILD [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    linkstatic = True,
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "memuse",
+    hdrs = ["memuse/memuse.h"],
+    linkstatic = True,
+    visibility = ["//visibility:public"],
+)
```
Comment (kkrol27) [inline lib/fsw/BUILD:13]: Define includes for these as well as opposed to explicit command line options?

File: src/fsw/FCCode/fswbase.bzl [hunk 4]
```diff
@@ -0,0 +1,34 @@
+def fswbase(name, defines, deps):
+    native.cc_library(
+        name = name,
+        srcs = native.glob(["**/*.cpp", "**/*.inl"]),
+        hdrs = native.glob(["**/*.hpp", "**/*.h"]),
+        linkstatic = True,
+        visibility = ["//visibility:public"],
+        copts = ["-Isrc/ -Isrc/fsw/ -Isrc/fsw/Devices -Ilib/common/libsbp/include -Ilib/common/concurrentqueue -Ilib/common/psim/include -Ilib/fsw/json  -Ilib/fsw/memuse"],
```
Comment (kkrol27) [inline src/fsw/FCCode/fswbase.bzl:8]: I think we can carry a lot of these includes through with the targets.

**Comment by kkrol27** [comment]
> Closing to simplify the structure. There's way too much code repetition that can be simplified by using Bazel `select`s.

Ah oops, I didn't refresh my page so didn't see that you had closed this lol

**Pull request closed without merging**